js/civs.js

```diff
@@ -1,18 +1,199 @@
+         uniqueUnitElite: civ.unique[1],
+         uniqueTechOne: civ.unique[2],
+         uniqueTechTwo: civ.unique[3],
+         monkPrefix: civ.monkPrefix});
+   
+   // Disable bldgs, units, techs.
+   if (civ.disabled) {
+      if (civ.disabled.buildings) builder.disableBuildings(civ.disabled.buildings);
+      if (civ.disabled.units) builder.disableUnits(civ.disabled.units);
+      if (civ.disabled.techs) builder.disableTechs(civ.disabled.techs);
+   }
 
-    let selectedCiv = civsConfig[name];
+   // Enable bldgs, units, techs.
+   if (civ.enabled) {
+      if (civ.enabled.buildings) builder.enableBuildings(civ.enabled.buildings);
+      if (civ.enabled.units) builder.enableUnits(civ.enabled.units);
+      if (civ.enabled.techs) builder.enableTechs(civ.enabled.techs);
+   }
 
-    let enabled = selectedCiv.enabled || {};
-    let disabled = selectedCiv.disabled || {};
-    let uniqueConfig = selectedCiv.unique || {};
-    if (selectedCiv.disableHorses) {
-        disableHorses();
-    }
+   if (civ.disableHorses) builder.disableHorses();
+   
+   return builder.build();
+}
+
+function civ(name) {
+   let selectedCiv = parseCiv(civsConfig[name]);
+   const id_regex = /(?<type>.+)_(?<id>([\d]+|unique_unit|elite_unique_unit))_(x|copy)/;
+
+   // On reset, we need to NOT enable things we're about to disable again.
+   SVG.select('.cross').each(function(i) {
+      // If fill opacity is already 0, then this is already enabled.
+      if (this.attr('fill-opacity') === 0) return;
+      
+      // Parse this element's ID number.
+      let elId = this.id();
+      const found = elId.match(id_regex);
+      if (!found) {
+         return;
+      };
+      let id = parseInt(found.groups.id);
+      let type = found.groups.type;
+
+      // If this ID is in the next civ's disable list, then don't enable it.
+      if (type === 'unit') {
+         if (selectedCiv.units.includes(id)) return;
+      } else if (type === 'building') {
+         if (selectedCiv.buildings.includes(id)) return;
+      } else if (type === 'tech') {
+         if (selectedCiv.techs.includes(id)) return;
+      }
+
+      // Enable the element.
+      this.attr({'fill-opacity': 0});
+   });
+
+   // Make sure the uniques are enabled.
+   enable([], [UNIQUE_UNIT, ELITE_UNIQUE_UNIT], []);
+   // Disable based on the civ object.
+   disable(selectedCiv.buildings, selectedCiv.units, selectedCiv.techs);
+   // Setup the unique objects.
+   unique([selectedCiv.uniqueConfig.uniqueUnit,
+     selectedCiv.uniqueConfig.uniqueUnitElite,
+     selectedCiv.uniqueConfig.uniqueTechOne,
+     selectedCiv.uniqueConfig.uniqueTechTwo], selectedCiv.monkPrefix);
+}
+
+// A civ object contains lists of buildings, techs, and units to explicitly
+// disable, as well as unique unit and tech config.
+class Civ {
+  constructor(buildings, techs, units, monkPrefix, uniqueConfig) {
+     this.buildings = buildings || [];
+     this.techs = techs || [];
+     this.units = units || [];
```

Comment:
Aren't those effectively lists of disabled entities?
```suggestion
     this.disabledBuildings = buildings || [];
     this.disabledTechs = techs || [];
     this.disabledUnits = units || [];
```

Reply:
Changed in the builder and object.